languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -28,11 +31,25 @@ def clear
       # @param name [String]
       # @raise [PolarFileExtensionError] if provided filename has invalid extension.
       # @raise [PolarFileNotFoundError] if provided filename does not exist.
-      def load_file(name)
+      def load_file(name) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         raise PolarFileExtensionError unless ['.pol', '.polar'].include? File.extname(name)
         raise PolarFileNotFoundError, name unless File.file?(name)
 
-        load_queue << name
+        hash = hash_file(name)
+
+        if @loaded_names.key?(name)
+          raise PolarRuntimeError, "File #{name} has already been loaded." unless @loaded_names[name] != hash
+
+          raise PolarRuntimeError, "A file with the name #{name}, but different contents,
+            has already been loaded."
```

Comment:
Actually on second thought I kinda like the readability a bit more if we reverse the order of these two `raise`s so that the conditional check mirrors the error text:

```suggestion
          raise PolarRuntimeError, "A file with the name #{name} but different contents
          has already been loaded." if loaded_names[name] != hash

          raise PolarRuntimeError, "File #{name} has already been loaded."
```

Reply:
nit: would prefer these were new error types instead of `PolarRuntimeError`s with inline messages